system/modules/task/actions/edit.php

```diff
@@ -206,12 +209,13 @@ function edit_POST($w) {
     }
     
     $task->fill($_POST['edit']);
+    $task->rate = abs($task->rate) == 0 ? NULL : $task->rate;
```

Comment:
Why check against the absolute value of rate, but not set the rate variable to the absolute value? Do we want negative rates?